rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -894,6 +897,26 @@ stream {
         location = {{ $authPath }} {
             internal;
 
+            {{ if $externalAuth.AuthCacheKey }}
+            set $cache_key '{{ $externalAuth.AuthCacheKey }}';
+
+            rewrite_by_lua_block {
+                local hash = ngx.sha1_bin(ngx.var.cache_key)
```

Comment:
You can embed `$externalAuth.AuthCacheKey` directly here as a string instead of introducing a variable.

Reply:
Oh I guess if you directly embed here then you'd have to implement concatenation of multiple variables yourself - this is good.

Reply:
If i embed `$externalAuth.AuthCacheKey` in a lua block i (as a user) can not use variables like `$uri` or `$cookie_name` via annotation or configMap.

If i use them before the lua block nginx replaces them with their values:
`$uri` becomes `/_external-auth-Lw` etc..

Reply:
yup, that's what I figured after posting my comment (hence the second comment as an update)

not asking for any change here if that wasn't clear in my second comment